src/hoc/getUserRole.tsx

```diff
@@ -0,0 +1,20 @@
+import { ROLES } from "@/constants/roles";
+import { useSession } from "next-auth/react";
+
+const GetUserRole = (condition: "isSupplier" | "isAdmin" | "isCommomUser") => {
+  const { data: session } = useSession();
+  const userRole = session?.user.role;
+
+  switch (condition) {
+    case "isCommomUser":
+      return userRole === ROLES.commomUser;
+    case "isSupplier":
+      return userRole === ROLES.supplier;
+    case "isAdmin":
+      return userRole === ROLES.admin;
+    default:
+      return false;
+  }
+};
+
+export default GetUserRole;
```

Comment:
```suggestion
const userRole = {
  const { data: session } = useSession();
  const  loggedUserRole = session?.user.role;
  
  const isSupplier = () => userRole === ROLES.supplier;
  const isCommonUser = () => userRole === ROLES.commonUser;
  const isAdmin = () => userRole === ROLES.admin;
};

export default userRoles;
```

What do you think?

In the page we would call:
`userRoles.isSupplier()` or `userRoles.isCommonUser()`

and then grow to:

`userRoles.isPremiumUser()` 
`userRoles.isExpiredPremiumUser()`
`userRoles.isNewUser()`
 etc....